CIP-XXXX/cardano-babbage.json

```diff
@@ -0,0 +1,1826 @@
+          "additionalProperties": false
+        },
+        {
+          "title": "NativeScript",
+          "type": "object",
+          "properties": {
+            "tag": {
+              "type": "string",
+              "enum": [
+                "native_script"
+              ]
+            },
+            "value": {
+              "$ref": "cardano-babbage.json#/definitions/NativeScript"
+            }
+          },
+          "required": ["tag", "value"],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "DataHash": {
+      "title": "DataHash",
+      "type": "string",
+      "format": "hex",
+      "pattern": "^([0-9a-f][0-9a-f]){32}$"
+    },
+    "TransactionOutput": {
+      "title": "TransactionOutput",
+      "type": "object",
+      "properties": {
+        "address": {
+          "$ref": "cardano-babbage.json#/definitions/Address"
+        },
+        "amount": {
+          "$ref": "cardano-babbage.json#/definitions/Value"
+        },
+        "plutus_data": {
+          "discriminator": {
+            "propertyName": "tag"
+          },
+          "oneOf": [
+            {
+              "type": "object",
+              "properties": {
+                "tag": {
+                  "enum": [
+                    "datum"
+                  ]
+                },
+                "value": {
+                  "$ref": "cardano-babbage.json#/definitions/PlutusData"
+                }
+              }
+            },
+            {
+              "type": "object",
+              "properties": {
+                "tag": {
+                  "enum": [
+                    "datum_hash"
+                  ]
+                },
+                "value": {
+                  "$ref": "cardano-babbage.json#/definitions/DataHash"
+                }
+              }
+            }
+          ]
+        },
+        "script_ref": {
+          "$ref": "cardano-babbage.json#/definitions/ScriptRef"
+        }
+      },
+      "required": [
+        "address",
+        "amount"
+      ]
+    },
+    "TransactionUnspentOutput": {
+      "title": "TransactionUnspentOutput",
+      "type": "object",
+      "properties": {
+        "input": {
+          "$ref": "cardano-babbage.json#/definitions/TransactionInput"
+        },
+        "output": {
+          "$ref": "cardano-babbage.json#/definitions/TransactionOutput"
+        }
+      },
+      "required": ["input", "output"],
+      "additionalProperties": false
+    },
+    "TransactionMetadatum": {
+      "title": "TransactionMetadatum",
+      "discriminator": {
+        "propertyName": "tag"
+      },
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "tag": {
+              "enum": [
+                "map"
+              ]
+            },
+            "contents": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "key": {
```

Comment:
Is this JSON format taken from somewhere or just here? It would be good for things that have JSON formats used by other tools like tx metadatums, plutus datums etc to stick to existing formats if possible.

Reply:
There is a plan to synchronize CSL with the schema, https://github.com/Emurgo/cardano-serialization-lib/issues/658

Reply:
cardano-node has a schema for metadatums and plutus datums. CSL and CML both support it. Note that for CSL it also has a slightly different auto-generated format if you use the auto-generated to_json() instead of the dedicated ones that let you use the cardano-node schemas.

overall I would recommend sticking as close as possible to be 1-to-1 with the .cddl binary spec to make it easier for tools in general to support it

Reply:
CIP57 too

Reply:
not sure if anyone uses them much but there are also cardano-node and cardano-cli formats for native scripts.

it would be good to use whatever is the most common formats for anything that has some kind of format already established, and then for the remainder to make it as close to 1-to-1 to the binary .CDDL format since thatś as close as we have to a standard format in general for things and CBOR and JSON can generally be converted directly as long as you're not using the features of CBOR not supported by JSON (complex keys e.g. in plutus/metadatums) or that aren't necessary like cbor tags/etc. Although the group choices e.g. certificates will need more thought on how to discriminate those

Reply:
with obvious considerations for making things more human readable e.g. hex string or bech32 for addresses makes much more sense than the raw bytes of the CBOR format

Reply:
@rooooooooob see this issue in cardano-cli for more context: https://github.com/IntersectMBO/cardano-cli/issues/570